sdk/keyvault/azure-security-keyvault-keys/src/test/java/com/azure/security/keyvault/keys/KeyClientTest.java

```diff
@@ -58,7 +63,8 @@ protected void createKeyClient(HttpClient httpClient, KeyServiceVersion serviceV
     }
 
     protected void createKeyClient(HttpClient httpClient, KeyServiceVersion serviceVersion, String testTenantId) {
-        HttpPipeline httpPipeline = getHttpPipeline(httpClient, testTenantId);
+        HttpPipeline httpPipeline = getHttpPipeline(buildSyncAssertingClient(httpClient == null
+            ? interceptorManager.getPlaybackClient() : httpClient), testTenantId);
         KeyAsyncClient asyncClient = spy(new KeyClientBuilder()
```

Comment:
We can remove this instance of `KeyAsyncClient` as it won't be used any longer. We should then use the `spy()` and `when()` methods on the `KeyClientImpl` from below.

Reply:
Adding comment to request changes.